security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -629,9 +638,16 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 		// Re-generate secret if it's expired.
 		if sc.shouldRotate(&secret) {
 			atomic.AddUint64(&sc.secretChangedCount, 1)
-			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
-			if sc.isTokenExpired(&secret) {
-				cacheLog.Debugf("%s token expired", logPrefix)
+			if UseRotationWithCert {
+				err := sc.fetcher.CaClient.Reconnect(true)
+				if err != nil {
+					cacheLog.Errorf("%s Connection Reset fails", logPrefix)
+					// Send the notification to close the stream if reconnection fails, so that client could re-connect with a new token.
+					sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
+					return true
+				}
+			} else if sc.isTokenExpired(&secret) {
+				cacheLog.Infof("%s token expired", logPrefix)
```

Comment:
Why move from debug -> info?

Reply:
Or maybe to error - and move this to pkg/agent, in the code that determines the options. 

There is no scenario where we can use an expired token - but maybe the code can fallback to existing certs.

Reply:
Yes I will set it as an error As when the token is expired the stream is closed so that it should throw out an error here